tests/utils/Plug.test.ts

```diff
@@ -0,0 +1,19 @@
+import Plug from '@/utils/Plug';
+
+describe(Plug.name, () => {
+  test('can plug and unplug', async () => {
+    const plug = new Plug();
+
+    // Calls are idempotent
+    await plug.plug();
+    await plug.plug();
+    await plug.plug();
+    expect(plug.isPlugged()).toBeTrue();
+
+    // Calls are idempotent
+    await plug.unplug();
+    await plug.unplug();
+    await plug.unplug();
+    expect(plug.isPlugged()).toBeFalse();
+  });
+});
```

Comment:
I think you should be using the `Lock`.

Reply:
Resolving this, discussion for it in the review comment below.